src/manifest.rs

```diff
@@ -129,9 +214,100 @@ impl CargoManifest {
     }
 }
 
+impl BuildMode {
+    fn from_table(table: Table) -> Result<BuildMode, Error> {
+        let mut build_mode = BuildMode::default();
+        for (key, value) in table {
+            match (key.as_str(), value.clone()) {
+                ("debug", Value::Table(t)) => {
+                    build_mode.debug = Some(WasmPackConfig::from_table(t)?)
+                }
+                ("profiling", Value::Table(t)) => {
+                    build_mode.profiling = Some(WasmPackConfig::from_table(t)?);
+                }
+                ("production", Value::Table(t)) => {
+                    build_mode.production = Some(WasmPackConfig::from_table(t)?);
+                }
+                (key, value) => Err(format_err!(
+                    "unexpected \
+                     `package.metadata.wasm_pack.build` key `{}` with value `{}`",
+                    key,
+                    value
+                ))?,
+            }
+        }
+        Ok(build_mode)
+    }
+}
+
+impl WasmPackConfig {
+    fn from_table(table: Table) -> Result<WasmPackConfig, Error> {
+        let mut wasm_pack_config = WasmPackConfig::default();
+        for (key, value) in table {
+            match (key.as_str(), value.clone()) {
+                ("debug", Value::Boolean(b)) => wasm_pack_config.debug = From::from(b),
+                ("features", Value::Array(a)) => {
```

Comment:
how does features work?
@ashleygwilliams  impl From\<Features\> for [CargoTomlFeaturesSection](https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section)?